src/main/java/com/google/devtools/build/lib/rules/cpp/CppModuleMapAction.java

```diff
@@ -77,7 +89,10 @@ public CppModuleMapAction(
       boolean compiledModule,
       boolean moduleMapHomeIsCwd,
       boolean generateSubmodules,
-      boolean externDependencies) {
+      boolean externDependencies,
+      OutputPathsMode outputPathsMode,
+      BiFunction<ImmutableMap<String, String>, String, ImmutableMap<String, String>>
+          modifyExecutionInfo) {
```

Comment:
This design of passing in a BiFunction, checking if it would result in one key SUPPORTS_PATH_MAPPING and then modifying the execution info by adding that key, looks fishy.

I'd be more comfortable passing in `ImmutableMap<String, String> executionInfo` and following SpawnAction example. Interning it and returning it in getExecutionInfo.

It will probably require a bit more internal testing, but it's also more future proof.

Reply:
I am now passing in the execution info map directly. I still need some extra logic to save a field (which would result in a larger instance size), PTAL.